### PR TITLE
League post

### LINKED ISSUE #11: API: Get all league posts
- User story #4

Requirements:

- #157 
- #159 
- #162 
- #164

### LINKED ISSUE #157: Database: Add LeaguePost Entity
User Stories: 

- #7 

Fields:

- id: Long
- numberOfPlayers: Integer
- numberOfComments: Integer
- leagueDetails: LeagueDetails
- description: String

### LINKED ISSUE #159: Database: Add LeagueDetails Entity
User Stories: 

- #5

Fields:

- id: Long
- name: String
- gameType: String (enum table?)
- timestamp: Date
- leaguePlayers: List\<User\>
- numberOfRounds: Integer
- numberOfFinishedRounds: Integer
- comments: List\<PostComment\>

### LINKED ISSUE #162: Database: Add LeaguePost Repository
User Stories: 

- #7

### LINKED ISSUE #164: Database: Add LeagueDetails Repository
User Stories: 

- #5

### PR DESCRIPTION
**Pull request by dudaskobendeguz**
- closes #11 
- closes #157
- closes #159
- closes #162 
- closes #164

### CODE REVIEW

File: springboot/src/main/java/com/codecool/terraformingmarsforum/controller/LeaguePostController.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("api/league-posts")
+public class LeaguePostController {
+    private final LeaguePostService leaguePostService;
+
+    @GetMapping
+    public ResponseEntity<List<LeaguePostModel>> getAll() {
+        return ResponseEntity.ok().body(leaguePostService.findAll());
```
Comment (benedekhalaj) [inline springboot/src/main/java/com/codecool/terraformingmarsforum/controller/LeaguePostController.java]: .ok() doesn't need body, you can put the data in the ok itself

File: springboot/src/main/java/com/codecool/terraformingmarsforum/sample_data/SampleDataInitializer.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import javax.annotation.PostConstruct;
+import java.util.Date;
+
+@Component
+@RequiredArgsConstructor
+public class SampleDataInitializer {
+
+    private final Logger logger = LoggerFactory.getLogger(SampleDataInitializer.class);
+    private final AppUserRepository appUserRepository;
+    private final CommentRepository commentRepository;
```
Comment (benedekhalaj) [inline springboot/src/main/java/com/codecool/terraformingmarsforum/sample_data/SampleDataInitializer.java]: Isn't it DataSampleInitializer? :D

File: springboot/src/main/java/com/codecool/terraformingmarsforum/service/LeaguePostService.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class LeaguePostService {
+
+    private final LeaguePostRepository leaguePostRepository;
+
+    public List<LeaguePostModel> findAll() {
+        List<LeaguePost> leaguePosts = leaguePostRepository.findAll();
+        return leaguePosts.stream().map(this::createLeaguePostModel).toList();
```
Comment (benedekhalaj) [inline springboot/src/main/java/com/codecool/terraformingmarsforum/service/LeaguePostService.java:21]: clean

**Review by benedekhalaj — APPROVED** [review]

File: springboot/src/main/java/com/codecool/terraformingmarsforum/service/LeaguePostService.java (hunk 3 above)
Comment (dudaskobendeguz) [reply springboot/src/main/java/com/codecool/terraformingmarsforum/service/LeaguePostService.java:21]: thanks  :safety_pin:

**Pull request merged**